tracer/src/Datadog.Trace/IAST/IastModule.cs

```diff
@@ -44,12 +44,22 @@ public IastModule()
 
     private static Scope? GetScope(Tracer tracer, string evidenceValue, IntegrationId integrationId, string vulnerabilityType, string operationName, Iast iast)
     {
-        if (!tracer.Settings.IsIntegrationEnabled(integrationId))
+        if (!iast.Settings.Enabled || !tracer.Settings.IsIntegrationEnabled(integrationId))
         {
             // integration disabled, don't create a scope, skip this span
             return null;
         }
 
+        var traceContext = (tracer.ActiveScope as Scope)?.Span?.Context?.TraceContext;
```

Comment:
It's technically possible for a user's (version-mismatched) span to be the `Tracer.ActiveScope`, so we should be checking the property `Tracer.InternalActiveScope`, which will always return a `Scope?` object from this module. Regardless of version mismatch, the returned `Scope?` object will have a root span with our automatic instrumentation, assuming it's a web server request.

```suggestion
        var traceContext = tracer.InternalActiveScope?.Span?.Context?.TraceContext;
```

Reply:
In the use case I am thinking about, we first open an automatic span (e.g. `aspnet.request`) and then a user opens a (version-mismatched) custom span. When this happens, the current logic would get the custom span but it's not of the Scope type so we can't look up the root span, and so the result is a new span to represent the vulnerability.

Even though I'd like to always get the root span, I think adding a new vulnerability span would be the most reliable way to add this information right now without hitting weird edge cases. If this new vulnerability span is too much overhead, we could see if there's a more reliable way to get the root span in more cases in the future